bin/deploy.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/sh -l
+
+COMMIT_BRANCH=''
+WEB_APP_URL=''
+APP_ENGINE_URL=''
+
+if [ -z "$1" ]
+then
+  echo "You did not specify a deploy environment, your choices are: production, develop, staging"
+  exit 1
+fi
+
+case $1 in
+
+  "staging")
+    COMMIT_BRANCH='webapp-staging'
+    WEB_APP_URL='https://simplenote-staging.go-vip.net'
+    APP_ENGINE_URL='https://staging.simplenote.com'
+    ;;
+
+  "develop")
+    COMMIT_BRANCH='webapp-develop'
+    WEB_APP_URL='https://simplenote-develop.go-vip.net'
+    APP_ENGINE_URL='https://develop.simplenote.com'
+    ;;
+
+  "production")
+    COMMIT_BRANCH='webapp'
+    WEB_APP_URL='https://simplenote.go-vip.co'
+    APP_ENGINE_URL='https://app.simplenote.com'
+    ;;
+
+  *)
+    exit 0
+    ;;
+esac
+
+# Install Node Modules
+npm install
+
+# Temporarily move config-local so it doesn't get overwritten
+mv config-local.json config-local.original.json
+
+# Setup config
+json_string=$(printf '{"app_id": "chalk-bump-f49", "app_key": "12345", "development": false, "wpcc_client_id": "0", "wpcc_redirect_url": "https://simplenote.com", "is_app_engine": true, "web_app_url": "%s", "app_engine_url": "%s"}' "$WEB_APP_URL" "$APP_ENGINE_URL")
+echo "$json_string" > config-local.json
+
+# Clean up old builds
+rm -rf dist
+
+# Build site
+NODE_ENV=production make build
+
+# Restore config-local
+rm config-local.json
+mv config-local.original.json  config-local.json
+
+# Checkout deploy branch
+if [ "$(git branch | grep $COMMIT_BRANCH)" ]
+then
+  git branch -D $COMMIT_BRANCH
+fi
+git checkout -b $COMMIT_BRANCH
+
+# Delete files not needed in production
+find . -maxdepth 1 ! -name '.git' ! -name 'dist' ! -name 'vip' ! -name 'config-local.json' -exec rm -rf {} ';'
+
+# Copy vip files to base path
+cp -r vip/* ./
+rm -rf vip
+
+# Add, commit, push
+touch .gitignore
+echo config-local.json >> .gitignore
+git add --all
+git commit -m "Build: $COMMIT_BRANCH [ci skip]"
```

Comment:
if we're wanting to deploy a given branch why are we building this and making a commit to that branch?
do we not have a way to inject a different config for the different environments?

seems like we ought to be able to do it like this…

```bash
git checkout webapp-develop
git reset --hard develop
```

Does something in the build process push itself out to VIP Go?

Reply:
When the head of the webapp, webapp-staging, and webapp-develop branches change VIP Go grabs the contents of the repo and pushes it to the server.

We do not have a way to inject a different config for the different environments.

I am not sure what the code suggestion achieves.

Reply:
the code suggestion would change the `HEAD` of those branches and should trigger web hooks to fire. the branches effectively would work like tags.

Reply:
VIP Go doesn't build the app it only grabs what's in the folder. https://github.com/Automattic/simplenote-electron/tree/webapp